.circleci/config.yml

```diff
@@ -228,6 +228,38 @@ jobs:
           name: 'Install and test postgres upgrade'
           command: 'install-and-test-ext --target check-upgrade --old-pg-version 11 --new-pg-version 12'        
 
+
+
+  test-10_check-citus-upgrade:
+    docker:
+      - image: 'heisenberg302/citusupgradetester:latest'
```

Comment:
I will change this before merging.

Reply:
We only need 1 postgres version to execute a citus upgrade but I test both `citus_initial_version pg10 -> citus_master pg10` and `citus_initial_version pg11 -> citus_master pg11`. For example, currently in `citusupgradetester` image we have citus artifacts built for:
```
citus v8.0.0 pg10
citus v8.1.0 pg10
citus v8.2.0 pg10
citus v8.3.0 pg10
citus v8.0.0 pg11
citus v8.1.0 pg11
citus v8.2.0 pg11
citus v8.3.0 pg11
```

They are already in the image therefore we dont spend any time installing them during testing. Otherwise we will need to:
- Download citus with the given version
- Install citus

Doing this for each combination is not fast. I have not recorded the time it takes to do these steps but from my observation it is worth generating the citus artifacts during build time of images.

There was a branch in the repository, which is https://github.com/citusdata/the-process/tree/citusUpgradeTester. I have created a PR now but our images have changed therefore the PR will need to be updated, yet you can still take a look at it to see what is going on there, https://github.com/citusdata/the-process/pull/31

Reply:
makes sense to package the citus artifacts, can imagine that will be faster compared to installing from source in the test.

But we don't need the pg11 + its citus versions when running the pg10 test, could we instead make 2 images? Separate images for `pg10` and `pg11`.

For the artifacts, would it make sense to instead have them in tar balls to download the debs from our repo during the build of the image? That way we still only have to download them once (image build time) but we have guarantees that we are actually testing to our released versions and it is probably easier keeping it up to date.

downloading debs: https://unix.stackexchange.com/questions/408346/how-to-download-package-not-install-it-with-apt-get-command

Reply:
Yeah technically we dont need pg11 while testing pg10. But I feel like having more images is harder to maintain. What is the downside of having pg11 in the image while testing pg10? It increases the size of image, but it is very little compared to the total size. And downloading images is not a bottleneck in our tests.

I think we can download artifacts from our repo. I am not sure how much of a change it is, but that can also work technically.